### PR TITLE
rgw: release unused callback argument

### PR DESCRIPTION
**Pull request by IlsooByun**
If _aio_operate_ succeeds, _BucketIndexAioArg_ is released from the callback function(_bucket_index_op_completion_cb_).
However, if _aio_operate_ fails, _BucketIndexAioArg_ is not released.

Signed-off-by: Ilsoo Byun <ilsoobyun@linecorp.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cls/rgw/cls_rgw_client.h [hunk 1]
```diff
@@ -104,6 +104,7 @@ class BucketIndexAioManager {
     if (r >= 0) {
       add_pending(arg->id, c, oid);
     } else {
+      arg->put();
```
Comment (tchaikov) [inline src/cls/rgw/cls_rgw_client.h]: does this also apply to `aio_operate(librados::IoCtx& io_ctx, const string& oid, librados::ObjectWriteOperation *op)` ?

Comment (IlsooByun) [reply src/cls/rgw/cls_rgw_client.h]: the aforementioned function put the op into the internal queue, and if it is processed, the callback function is called, inside of which the callback argument is released. But if the error occurs before being put into the queue, it just returns with error code, and there is no chance to release the resources.

Comment (IlsooByun) [reply src/cls/rgw/cls_rgw_client.h]: @tchaikov I'm sorry I got it wrong. You are right. Same for aio_operate for writing. I'll push it again.

Comment (tchaikov) [reply src/cls/rgw/cls_rgw_client.h]: thanks!

**Comment by theanalyst** [comment]
passed qa suite run: http://pulpito.ceph.com/abhi-2020-01-24_16:05:12-rgw-wip-abhi-testing-2020-01-24-1135-distro-basic-smithi/ (which had s3-tests failure because of not being based on ceph-master)
rerun: http://pulpito.ceph.com/abhi-2020-01-27_10:30:43-rgw-wip-abhi-testing-2020-01-24-1135-distro-basic-smithi/

**Pull request merged**